From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Alphi <alphi@starchaser.local>
Date: Mon, 21 Sep 2026 11:20:35 +0200
Subject: [PATCH] fix(x11): add nullptr check for xGetImage in
 createSnapshotOfNativeWindow

Prevents hard crash on Linux X11 when taking window snapshots during window state transitions or when XGetImage returns NULL.
---
 .../native/juce_XWindowSystem_linux.cpp       | 20 +++++++++++--------
 1 file changed, 12 insertions(+), 8 deletions(-)

diff --git a/modules/juce_gui_basics/native/juce_XWindowSystem_linux.cpp b/modules/juce_gui_basics/native/juce_XWindowSystem_linux.cpp
index 93cff29..c29a4e6 100644
--- a/modules/juce_gui_basics/native/juce_XWindowSystem_linux.cpp
+++ b/modules/juce_gui_basics/native/juce_XWindowSystem_linux.cpp
@@ -4427,14 +4427,18 @@ Image createSnapshotOfNativeWindow (void* window)
         return 1.0;
     }();
 
-    auto image = Image { new XBitmapImage { X11Symbols::getInstance()->xGetImage (display,
-                                                                                  (::Drawable) window,
-                                                                                  0,
-                                                                                  0,
-                                                                                  ww,
-                                                                                  wh,
-                                                                                  AllPlanes,
-                                                                                  ZPixmap) } };
+    auto* xImage = X11Symbols::getInstance()->xGetImage (display,
+                                                         (::Drawable) window,
+                                                         0,
+                                                         0,
+                                                         ww,
+                                                         wh,
+                                                         AllPlanes,
+                                                         ZPixmap);
+    if (xImage == nullptr)
+        return {};
+
+    auto image = Image { new XBitmapImage { xImage } };
     return image.rescaled ((int) ((double) ww / scale), (int) ((double) wh / scale));
 }
 
